spec/session_ssl_spec.rb

```diff
@@ -214,14 +213,14 @@
     expect(YAML::load(response).header).to_not include('cookie')
   end
 
-  it "should ignore a wrong Content-Length when asked to" do
+  xit "should ignore a wrong Content-Length when asked to" do
     expect {
       @session.ignore_content_length = true
       @session.get("/wrongcontentlength")
     }.to_not raise_error
   end
 
-  it "should fail by default with a Content-Length too high" do
+  xit "should fail by default with a Content-Length too high" do
```

Comment:
This test fails because we do not set the correct `Content-Length` in the response in the test server? Wouldn't it be better to find out why it fails?

Reply:
I managed to make this test pass either by setting protocol version to 1.0 or by passing `Connection: close` header:
```ruby
@session.http_version = 'HTTPv1_0'
```
Or:
```
@session.get("/wrongcontentlength", { 'Connection' => 'close' })
```
Otherwise, Puma will hold keep-alive connection and eventually request will fail due to timeout. I did the same fix for non-ssl specs too. 

However, it works only for libcurl compiled against OpenSSL. At Travis CI, libcurl is compiled against GnuTLS and request fails with such an error:
```
<Patron::Error: GnuTLS recv error (-9): A TLS packet with unexpected length was received.>
```
I guess it's reasonable. Sending wrong content-length is a protocol violation and GunTLS just appears to be more strict than OpenSSL.

Reply:
Gotcha. Then we should remove those tests IMO as they test divergent behavior for which kno "golden master" outcome can be established.

Reply:
OK